kolibri/plugins/coach/assets/src/views/exams-page/index.vue

```diff
@@ -1,22 +1,243 @@
 <template>
 
   <div>
-    <h1>{{ $tr('header') }}</h1>
+    <h1>{{ currentClass.name }} {{ $tr('exams') }}</h1>
+    <ui-radio-group
+      :name="$tr('examFilter')"
+      :label="$tr('show')"
+      :options="filterOptions"
+      v-model="filterSelected"
+    />
+    <router-link :to="createExamPage">
+      <ui-button
+        type="primary"
+        color="primary"
+        :raised="true"
+        icon="add">
+        {{ $tr('newExam') }}
+      </ui-button>
+    </router-link>
+    <table v-if="exams.length">
+      <thead>
+        <tr>
+          <th class="col-icon"></th>
+          <th class="col-title">{{ $tr('title') }}</th>
+          <th class="col-visibility">{{ $tr('visibleTo') }}</th>
+          <th class="col-action">{{ $tr('action') }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <exam-row
+          v-for="exam in filteredExams"
+          :examId="exam.id"
+          :examTitle="exam.title"
+          :examActive="exam.active"
+          :examDate="exam.dateCreated"
+          :examVisibility="exam.visibility"
+          :classId="currentClass.id"
+          :className="currentClass.name"
+          :classGroups="[]"
+          @changeExamVisibility="openChangeExamVisibilityModal"
+          @activateExam="openActivateExamModal"
+          @deactivateExam="openDeactivateExamModal"
+          @previewExam="openPreviewExamModal"
+          @viewReport="routeToExamReport"
+          @renameExam="openRenameExamModal"
+          @deleteExam="openDeleteExamModal"
+        />
+      </tbody>
+    </table>
+    <p v-else class="center-text"><strong>{{ $tr('noExams') }}</strong></p>
+
+    <activate-exam-modal
```

Comment:
@christianmemije iirc, the modal doesn't exist until it's called? If that understanding is correct, there would be code to create a modal inside of each row, but it wouldn't impede on performance (if that's what you're arguing for).

But I am a little conflicted regarding whether we should keep it outside or not because of where the call to open the modal happens